docs/components_page/api_doc.py

```diff
@@ -1,45 +1,49 @@
+import re
+
+import dash_bootstrap_components as dbc
+import dash_core_components as dcc
 import dash_html_components as html
-from dash.development._py_components_generation import (
-    filter_props,
-    js_to_py_type,
-    reorder_props,
-)
-
-
-def ApiDoc(component_metadata, component_name=None):
-    component_props = component_metadata.get("props", {})
-    return html.Div(
-        ArgumentsList(component_props, component_name),
-        className="api-documentation",
-    )
 
 
-def ArgumentsList(component_props, component_name):
-    if component_name is not None:
-        heading = f"Keyword arguments for {component_name}"
-    else:
-        heading = "Keyword arguments"
-    component_props = reorder_props(filter_props(component_props))
-    arguments = []
-    for name, metadata in component_props.items():
-        arguments.append(Argument(name, metadata))
-    if not arguments:
-        return []
-    return [
-        html.H4(heading, className="mt-5 mb-2"),
-        html.Ul(arguments, className="list-unstyled"),
+def ComponentReference(component_name):
+    component = getattr(dbc, component_name)
+    component_doc = component.__doc__
+
+    return_div = [
+        dcc.Markdown("### Keyword arguments for {}".format(component_name))
     ]
 
+    docs = component_doc.split("Keyword arguments:")[-1]
+
+    # formats code blocks that includes square brackets
+    docs = docs.replace("[", "\[").replace("]", "\]")
+    verbatim_regex = r"`((\\\[)((.|\n)*?)(\\\]))`"
+    docs = re.sub(re.compile(verbatim_regex), r"`[\3]`", docs)
```

Comment:
I don't think efficiency is really an issue here, but each of these `compile` statements could be moved out of the function so that they only run once, not each invocation of the function. E.g.

```python
VERBATIM_PATTERN = re.compile(r"`((\\\[)((.|\n)*?)(\\\]))`")

def ComponentReference(component_name):
    ...
    docs = VERBATIM_PATTERN.sub(r"`[\3]`", docs)
```

Reply:
Good idea - I moved `api_doc.py` into `markdown_parser.py` to make that easier to do.